### PR TITLE
Feature/new pme

### PR DESCRIPTION
**Pull request by jvoytek**
https://trello.com/c/BXWgHEqu/1558-10-850-implement-the-user-interface-code-for-a-zotero-based-bookmarklet

### CODE REVIEW

File: inject.js [hunk 1]
```diff
@@ -8064,15 +8020,16 @@ var BookmarkletFrame = function(url, width, height) {
 	height = Math.min(windowHeight-10, height);
 	width = Math.min(windowWidth-10, width);
 
-	this._dimmer = parentDoc.createElement("div");
+	/*this._dimmer = parentDoc.createElement("div");
```
Comment (aepheus) [inline inject.js]: remove?

File: inject.js [hunk 2]
```diff
@@ -8088,10 +8045,14 @@ var BookmarkletFrame = function(url, width, height) {
 	var frameElementStyle = window.frameElement.style;
 	for(var i in cssBookmarkletFrame) frameElementStyle[i] = cssBookmarkletFrame[i];
 	frameElementStyle.display = "block";
-	frameElementStyle.margin = "-"+height/2+"px 0 0 -"+width/2+"px";
+	//frameElementStyle.margin = "-"+height/2+"px 0 0 -"+width/2+"px";
```
Comment (aepheus) [inline inject.js:8044]: remove?

File: inject.js [hunk 3]
```diff
@@ -8088,10 +8045,14 @@ var BookmarkletFrame = function(url, width, height) {
...
 	frameElementStyle.width = width+"px";
-	frameElementStyle.height = height+"px";
-	if(url) this._appendFrameTo.appendChild(this._frame);
+	//frameElementStyle.height = height+"px";
```
Comment (aepheus) [inline inject.js:8046]: remove?

File: inject.js [hunk 4]
```diff
@@ -8115,7 +8076,7 @@ translate.setHandler("translators", function(obj, translators) {
 
 	if(translators && translators.length) {
 		if(translators[0].runMode === Zotero.Translator.RUN_MODE_IN_BROWSER) {
-			Zotero.ProgressWindow.changeHeadline("Saving Item...");
+			Zotero.ProgressWindow.changeHeadline("Analyzing...");
 		} else if(translators[0].runMode === Zotero.Translator.RUN_MODE_ZOTERO_SERVER) {
 			Zotero.ProgressWindow.changeHeadline("Saving via Server...");
```
Comment (aepheus) [inline inject.js:8077]: It's polish, but we should remove code that will never execute (like this)

File: inject.js [hunk 5]
```diff
@@ -8125,28 +8086,90 @@ translate.setHandler("translators", function(obj, translators) {
...
+			function(statusCode, response) {
+			});
+		//translate.translate();
 
-		new Zotero.ProgressWindow.ErrorMessage("noTranslator");
-		Zotero.ProgressWindow.startCloseTimer(8000);
-		cleanup();
+		//new Zotero.ProgressWindow.ErrorMessage("noTranslator");
+		//Zotero.ProgressWindow.startCloseTimer(8000);
+		//cleanup();
```
Comment (aepheus) [inline inject.js]: remove?

File: inject.js [hunk 6]
```diff
@@ -8125,28 +8086,90 @@ translate.setHandler("translators", function(obj, translators) {
...
+		originalCallback(items);
+	} else {
+		originalCallback = callback;
+		var payload = {items:items};
+		Zotero.API.createSelection(payload,function(){console.log("completed Zotero.API.createSelection")});
+	}
+
+
+
+	/*
```
Comment (aepheus) [inline inject.js]: remove?

File: inject.js [hunk 7]
```diff
@@ -8196,11 +8219,12 @@ Zotero.Messaging.addMessageListener("translate", function(data, event) {
 });
 Zotero.Messaging.addMessageListener("selectDone", function(returnItems) {
 	// if no items selected, close save dialog immediately
-	if(!returnItems || Zotero.Utilities.isEmpty(returnItems)) {
+	/*if(!returnItems || Zotero.Utilities.isEmpty(returnItems)) {
```
Comment (aepheus) [inline inject.js]: remove?

**Comment by dash-** [comment]
Unsubscribe please :-)

Thanks!
David

**Comment by aepheus** [comment]
@dash- I think you have to unsubscribe yourself up at the top. "Unwatch"

**Pull request merged**